docs/arcadia-users-group/20231104-testing-concepts/lesson.md

```diff
@@ -0,0 +1,573 @@
+For instance, an input file for version X of a workflow is processed, and the output file is saved, typically online.
+While developing version Y, the test suite automatically fetches the output for version X, processes the same input file for version Y, and contrasts the two output files.
+Any significant discrepancies trigger a test failure.
+Regression tests can identify failures missed by integration and unit tests.
+Each project may adopt a slightly varied approach to regression testing, based on its software requirements.
+Testing frameworks aid in constructing regression tests but don’t provide additional sophistication beyond the discussed concepts.
+
+## Continuous integration
+
+Continuous integration makes running tests as easy as possible by integrating the test suite into the development process. 
+Every time a change is made to the repository, the continuous integration system builds and checks that code.
+
+Based on instructions you provide, a continuous integration server can:
+
+- check out new code from a repository
+- spin up instances of supported operating systems (i.e. various versions of OSX, Linux, Windows, etc.).
+- spin up those instances with different software versions (i.e. python 2.7 and python 3.0)
+- run the build and test scripts
+- check for errors
+- report the results.
+
+Since the first step the server conducts is to check out the code from a repository, we'll need to put our code online to make use of this kind of server.
+
+### Set Up a Mean Git Repository on GitHub
+
+Our `mean.py` `test_mean.py` files can be the contents of a repository on GitHub.
+
+* Go to GitHub and [create a repository](https://github.com/new) called aug-mean. Do this in your own user account and don't add any files to the new repo (no README/LICENSE, etc.).
+* Turn the `aug-mean` directory that we've been working in on your computer into a git repository following the "…or create a new repository on the command line" instructions on GitHub:
+```
+echo "# aug-mean" >> README.md
+git init
+git add README.md
+git commit -m "first commit"
+git branch -M main
+git remote add origin git@github.com:yourusername/aug-mean.git
+git push -u origin main
+```
+* Create a new branch (`git checkout -b yourinitials/init`).
+* Use git to `add`, `commit`, and `push` the two files `mean.py` and `test_mean.py` to GitHub.
+
+### GitHub Actions
+
+[GitHub Actions](https://github.com/features/actions) is a continuous integration service provided by GitHub.
+It's integrated directly into GitHub repositories and does not require additional accounts or external services.
+
+To use GitHub Actions, create a directory called `.github` and within it, create another directory called `workflows`.
+
+```
+mkdir -p .github/workflows
+```
+
+Inside the `workflows` directory, you can create a YAML file (e.g. `ci.yml`) to define your continuous integration process:
+
+```
+name: Continuous Integration
```

Comment:
I'd probably name it more testing focused

Reply:
fixed